source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -159,22 +159,36 @@ def create_next_kraken_request(request, responses):
     """
     create a new request dict to call the next (resp previous for non clockwise search) journeys in kraken
 
-    to do that we find the soonest departure (resp tardiest arrival) we have in the responses
-    and add (resp remove) one minute
+    to do that we find the best journey (soonest arrival, then shortest journey) found in the responses
+    and add one second to it's departure.
+    (if anticlockwise, remove one second to the arrival of the journey with latest departure, then shortest)
     """
-    one_minute = 60
+    one_second = 1
+    best_arrival = best_departure = None
     if request['clockwise']:
-        try:
-            soonest_departure = min(j.departure_date_time for r in responses for j in r.journeys if _has_pt(j))
-        except ValueError:
+        for r in responses :
```

Comment:
seems like dupplicated code, I think it would be nice to factorize it (use a function for the purpose to be clearer ?